src/actions/mintArveaweNFT.ts

```diff
@@ -0,0 +1,266 @@
+import BN from 'bn.js';
+import { calculate } from '@metaplex/arweave-cost';
+import { LAMPORTS_PER_SOL, PublicKey } from '@solana/web3.js';
+import crypto from 'crypto';
+import { Connection } from '../Connection';
+import { MetadataJson } from '../types';
+import { Wallet } from '../wallet';
+import { ArweaveStorage, ArweaveUploadResult } from '../providers';
+import { PayForFiles } from '../programs';
+import { prepareTokenAccountAndMintTx } from './shared';
+import { Metadata, Creator, MetadataDataData, CreateMetadata } from '../programs/metadata';
+import { sendTransaction } from './transactions';
+import { updateMetadata } from './updateMetadata';
+import { createMasterEdition } from './createMasterEdition';
+
+export enum ENftProgress {
+  none,
+  minting,
+  preparing_assets,
+  signing_metadata_transaction,
+  sending_transaction_to_solana,
+  waiting_for_initial_confirmation,
+  waiting_for_final_confirmation,
+  uploading_to_arweave,
+  updating_metadata,
+  signing_token_transaction,
+}
+
+export async function convertFile2Hash(file: File): Promise<Buffer> {
```

Comment:
Is this intended to be exported? Same for other functions that are utils.

Reply:
Better move these to utils

Reply:
I've updated `getFileHash`` from `utils/crypto` for this purposes.